exercises/practice/leap/.approaches/using-div/content.md

```diff
@@ -0,0 +1,52 @@
+# Using the `div` instruction
+
+```asm
+## Registers
+
+# | Register | Usage     | Type    | Description                                      |
+# | -------- | --------- | ------- | ------------------------------------------------ |
+# | `$a0`    | input     | integer | year to check                                    |
+# | `$v0`    | output    | boolean | input is leap year (`0` = `false`, `1` = `true`) |
+# | `$t0-9`  | temporary | any     | used for temporary storage                       |
+
+.globl is_leap_year
+
+is_leap_year:
+        li      $v0, 0
+
+        li      $t0, 4
+        div     $a0, $t0
+        mfhi    $t0
+        bnez    $t0, end
+
+        li      $t0, 100
+        div     $a0, $t0
+        mfhi    $t0
+        bnez    $t0, set_leap
+
+        li      $t0, 400
+        div     $a0, $t0
+        mfhi    $t0
+        bnez    $t0, end
+
+set_leap:
+        li      $v0, 1
+end:
+        jr      $ra
+```
+
+Equivalent C code:
+
+```c
+int is_leap_year(int year) {
+    return year % 4 == 0 && year % 100 != 0 || year % 400 == 0;
```

Comment:
```
return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
```

Reply:
I changed it, however in this case it is actually equivalent; but I think you are correct that it is easier to understand with proper grouping.